### PR TITLE
Upgrade Matomo from 3.11.0 to 4.2.1

### PR DESCRIPTION
**Pull request by LiquidITGuy**
Upgrade database dependencies
remove useless params and useless root password
upgrade Matomo from 3.11.0 to 4.2.1

### ☑️ Self Check before Merge

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [x] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [x] Icon is added as a png file to the logos directory.

### CODE REVIEW

File: public/v4/apps/matomo.yml [hunk 1]
```diff
@@ -8,11 +8,11 @@ services:
             MYSQL_DATABASE: matomo
             MYSQL_USER: matomo
             MYSQL_PASSWORD: $$cap_db_pass
-            MYSQL_ROOT_PASSWORD: $$cap_db_root_pass
```
Comment (githubsaturn) [inline public/v4/apps/matomo.yml:11]: Why is this removed? Isn't it needed for mysqldump?
To make it easier for the user, we can use `$$cap_gen_random_hex(18)` as the default value.

**Review by githubsaturn — COMMENTED** [review]
Thanks! Just one minor comment.

**Comment by LiquidITGuy** [comment]
Thanks for the review

I removed those lines cause they are not on the matomo docker compose sample in the last version but it can be a misunderstanding by me ?

For the size of the random password I'll fix it :)

File: public/v4/apps/matomo.yml (hunk 1 above)
Comment (LiquidITGuy) [reply public/v4/apps/matomo.yml:11]: I think you can just use the matomo users (who already have all the right on the table on the database)
I removed the usage of the root account to prevent missuse of root account but I can put it back if you prefear ?

Comment (LiquidITGuy) [reply public/v4/apps/matomo.yml:11]: Ps : I'm not sure about the fact the user can use mysql_dump, I'll check the matomo documentation for dump

Comment (githubsaturn) [reply public/v4/apps/matomo.yml:11]: I see. I didn't know there is an existing user that has access to all the tables. That should be fine then.
